Feature/homework26

EvgeniiKovalev opened the pull request:
ДЗ 26
1. Переделаны роли с перечисления на справочник (ин мемори провайдер адаптирован под новые роли)
2. Реализован работа приложения через JDBC провайдер

EvgeniiKovalev committed:
ДЗ23 базовый проект

EvgeniiKovalev committed:
ДЗ23 готово
1. Реализована возможность отправки личных сообщений: если клиент пишет «/w tom Hello», то сообщение Hello должно быть отправлено только клиенту с ником tom.
2. Выполнен рефакторинг обработки сообщений клиента на стороне сервера

EvgeniiKovalev committed:
ДЗ23 готово
1. Реализована возможность отправки личных сообщений: если клиент пишет «/w tom Hello», то сообщение Hello должно быть отправлено только клиенту с ником tom.
2. Выполнен рефакторинг обработки сообщений клиента на стороне сервера

EvgeniiKovalev committed:
ДЗ24 отключается админ отправивший команду /kick :)

EvgeniiKovalev committed:
ДЗ24
вылечил отключение админа отправившего команду /kick
но после отключения клиента по kick он безуспешно пытается отправить на сервер

EvgeniiKovalev committed:
ДЗ24
исправлено при помощи CountDownLatch - после отключения клиента по kick он безуспешно пытается отправить сообщение на сервер

EvgeniiKovalev committed:
ДЗ24 добавил pom файлы модулей в maven

EvgeniiKovalev committed:
ДЗ24 Класс User теперь внешний, добавлена роль по умолчанию -  USER

EvgeniiKovalev committed:
Merge remote-tracking branch 'origin/Feature/homework24' into Feature/homework24

EvgeniiKovalev committed:
ДЗ24
1. исправил pom файлы, удалил iml файлы из проекта,
    чтобы модуль чата можно было публиковать в git
    закомментирован блок определиния модуля чата в головном pom.xml
2. пересоздал модули чата
    chat2025_03
    client
    server
3. для модулей client и server указал  корневые папки моделей в качестве корневых папок исходников
в pom.xml модулей client и server

4. исправил замечания к pr

EvgeniiKovalev committed:
Merge remote-tracking branch 'origin/Feature/homework24' into Feature/homework24

EvgeniiKovalev committed:
ДЗ26 init

EvgeniiKovalev committed:
ДЗ26
1. Переделаны роли с перечисления на справочник (ин мемори провайдер адаптирован под новые роли)
2. Реализован работа приложения через JDBC провайдер

RomanVoronovskiy commented on src/main/java/ru/otus/java/basic/homeworks/homework26/chat2025_03/server/PostgresAuthenticationProvider.java:
```diff
@@ -0,0 +1,332 @@
+    }
+
+    @Override
+    public void initialize() {
```
выглядит громоздко, уж лучше вынести вообще в отдельный класс , сделать поля статиками и присвоить на месте, потом просто добавить импорт и использовать

RomanVoronovskiy commented on src/main/java/ru/otus/java/basic/homeworks/homework26/chat2025_03/server/PostgresAuthenticationProvider.java:
```diff
@@ -0,0 +1,332 @@
+    }
+
+    int saveUserTable(User user, Connection connection) throws SQLException {
+        int user_id  = user.getId();
```
camelCase(в ряде случаев есть ещё, но подсвечиваю как общий)

EvgeniiKovalev committed:
ДЗ26 исправлены замечания:
1. стат. константы вынесены в класс Constants
2. стиль кода(camelCase, отступы и т.д.)

EvgeniiKovalev committed:
ДЗ26 добавил файл для проверки стиля кода
